kedro/framework/project/__init__.py

```diff
@@ -63,6 +63,9 @@ class _ProjectSettings(LazySettings):
         "CONFIG_LOADER_CLASS", default=_get_default_class("kedro.config.ConfigLoader")
     )
     _CONFIG_LOADER_ARGS = Validator("CONFIG_LOADER_ARGS", default={})
+    _DATA_CATALOG_CLASS = Validator(
```

Comment:
Can we actually use `_IsSubclassValidator` like we do for `_SESSION_STORE_CLASS`? The config loader should also follow the same pattern - feel free to include it here or in a separate PR, whichever is easier for you.

Reply:
I changed `_CONTEXT_CLASS` to use `_IsSubclassValidator` as well